src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/OptionsTests.cs

```diff
@@ -171,15 +170,24 @@ public static void WhenAddingContextTypeInfoResolverAsContextOptionsAreSameAsOpt
         }
 
         [Fact]
-        public static void TypeInfoResolverCannotBeSetAfterContextIsSetThroughTypeInfoResolver()
+        public static void WhenAddingContext_SettingResolverToNullThrowsInvalidOperationException()
+        {
+            var options = new JsonSerializerOptions();
+            options.AddContext<JsonContext>();
+            Assert.Throws<InvalidOperationException>(() => options.TypeInfoResolver = null);
```

Comment:
Won't this also prevent workaround you added here? https://github.com/dotnet/runtime/issues/71716

I think we should not lock options when assigning them in the context although it might make JsonSerializerContext weird since you can't change options later but I think it's ok and it should be rather rare that someone would want to re-assign (other than copy ctor).